### PR TITLE
ContextSearchForString

### PR DESCRIPTION
**Pull request by Itay4**
Regarding issue https://github.com/demisto/server/issues/9404
Note that it depends on fix done to flattenFields in https://github.com/demisto/content/pull/936

### CODE REVIEW

File: Scripts/ContextSearchForString.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+commonfields:
+  id: bb2d479b-2bd4-40cb-84ac-9d7b620f060e
```
Comment (aviadl) [inline Scripts/ContextSearchForString.yml]: ID incorrect

File: Scripts/ContextSearchForString.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  version: 72
```
Comment (aviadl) [inline Scripts/ContextSearchForString.yml]: Version incorrect

File: Scripts/ContextSearchForString.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  return searchString(contextObject);
+
+
+type: javascript
+tags: []
+comment: |
+  Searches for string in a path in context. If path is null, string will be searched in full context.
+enabled: true
+args:
```
Comment (aviadl) [inline Scripts/ContextSearchForString.yml]: Missing output

File: Scripts/ContextSearchForString.yml (hunk 1 above)
Comment (dorsha) [reply Scripts/ContextSearchForString.yml]: id should be the same as the name

File: Scripts/ContextSearchForString.yml (hunk 2 above)
Comment (dorsha) [reply Scripts/ContextSearchForString.yml]: version should be -1

File: Scripts/script-ContextSearchForString.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+script: |+
+  var path = args.path;
+  var str = args.str;
+  var flat={};
+
+  function searchString(obj) {
+      flattenFields(obj,undefined,flat);
+      keysArr = Object.keys(flat);
+      for (i = 0; i < keysArr.length - 1; i++) {
+          if (flat[keysArr[i]] == str) {
```
Comment (Spivaka) [inline Scripts/script-ContextSearchForString.yml]: ===

Comment (Spivaka) [reply Scripts/script-ContextSearchForString.yml]: read about differences between == and === in javascript 
https://stackoverflow.com/questions/523643/difference-between-and-in-javascript

**Pull request merged**